src/ucs/memory/rcache.c

```diff
@@ -987,17 +1034,33 @@ static ucs_status_t ucs_rcache_global_list_add(ucs_rcache_t *rcache)
     atfork_installed = 1;
 
 out_list_add:
-    ucs_list_add_tail(&ucs_rcache_global_list, &rcache->list);
+    if (ucs_list_is_empty(&ucs_rcache_global_context.list)) {
```

Comment:
would we skip adding to global list if pthread_atfork in line 1027 failed?

Reply:
Yes, but it is also the case in `master` (I didn't change that).

Reply:
before this PR it made sense, since global list was only for atfork() hook. now it's also needed for cleanup